### ISSUE TITLE
[FEATURE] Status & Label Cleanup

### ISSUE THREAD

**Issue by aerugo** [cobudget/cobudget#923]
We need to change how bucket status and labels work. This issue describes all changes necessary for this update.

## New settings

First of all, we will need to introduce two new settings. 
Two new boolean switches will be added to Settings -> Funding.

* Co-creators can open buckets for funding: YES / NO [Default: NO]
   *  When this is set to YES, a co-creator of the bucket can see and click the "Open for funding button". When it is set to NO, only moderators and admins can see it and open the bucket for funding. 
* Co-creators can edit their bucket during funding: YES/NO [Default: NO] 
   * When set to NO, co-creators cannot edit their bucket until bucket has reached stage "funded"

Implement these new settings. Also, add toast popups that notify the user that they cannot edit their bucket since funding has opened when they try to click the edit pen symbol. Copy for the toast should be: "Funding has started, and you cannot edit your bucket. Please contact a moderator or admin for help."

## New UX element

A new info box, placed under the bucket creation date. This is not always visible, hide if there is no message to show for the user that is viewing the bucket.

<img width="522" alt="Screenshot 2023-01-15 at 02 04 10" src="https://user-images.githubusercontent.com/7785081/212503894-01da48ef-d98e-47a3-b846-1a60467b8fa0.png">

Infobox design can be re-used from create-group upgrade info box.

<img width="397" alt="Screenshot 2023-01-15 at 02 01 42" src="https://user-images.githubusercontent.com/7785081/212503869-507880cc-8848-4adc-b5f4-fc713f63ff7e.png">

## New bucket property

Add the new bucket property bool "ready for funding". This is **not** the same thing as "open for funding".

## Status, labels, and bucket lifecycle

A bucket moves through 5 stages in its lifecycle. 
Draft -> Idea -> Funding -> Funded -> Completed

### Draft
_All new buckets start in this stage._
_Buckets can also reach this stage by a co-creator, admin or moderator clicking "Unpublish"._

**Label name:** Draft
**Label color:** Grey

**Action buttons:** Publish

**Info box text:**
For co-creators: "Publish your bucket to move it to the idea phase."

**Notes:** _Draft buckets should not have the "Open for funding" action button._

### Idea
_When a bucket is published (by clicking the Publish action button), it progresses to Idea status._

**Label name:** Idea
**Label color:** Yellow

**Action buttons:**
  *Co-creators, admins, and moderators:* Unpublish
  * If the round setting allows co-creators to open for funding, show "Open for funding" action button to co-creators, moderators, and admins
  * Else:
     * to co-creators show action button “Ready for funding” / "Mark as not ready". This button flips the "ready for funding"-bool.
     * to admins show action button: "Open for funding"

**Info box text:**
For co-creators: 
  * If the round setting allows co-creators to open for funding, show "To start collecting funds for this bucket, open it for funding."
  * else, show "This bucket is waiting for an admin to open it for funding."
 
For admins:
  * Show "Ready to be opened for funding" or  "Not ready for funding yet"

For all regular users (both logged in and not):
  * Show "This bucket is waiting to be opened for funding."

### Funding
_When “Open for funding” is clicked, the bucket moves to funding stage_
_Buckets can also reach the funded stage from the Funded stage through the "Re-open for funding" button.

**Label name:** Funding
**Label color:** Blue

**Action buttons:** 
  * Signed-in members of round: Fund
  * If bucket has reached minimum funding, but not maximum: Accept and close
       * Pop-up: "Are you sure you want to accept the funding, even if you have not reached your max goal? You will need to contact an admin to open the bucket for funding again."

**Info box text:**
For co-creators: 
  * If the round settings allow co-creators to edit their budget during funding: "To accept funding, funds raised must match your budget. Edit your budget below to match the funding you received."
  * Else: “To accept funding, funds raised must match your budget. Only admins can edit budgets after funding starts.”

### Funded
_When a bucket reaches its minimum goal, or its maximum goal, or when co-creator clicks "Accept and close", the bucket moves to "funded" stage._

**Label name:** Funded
**Label color:** Green

**Action buttons:** 
  * For co-creators: “Mark as completed”
     *  Here, have a popup "Are you sure you would like to mark this bucket as completed? This can't be undone."
  * For admins and moderators: “Re-open for funding”

**Info box text:**
For co-creators: "When you have used your funding, you can mark this bucket as completed. Remember that it is usually appreciated to write a short message in the comments about how it turned out."

### Completed

**Label name:** Completed
**Label color:** Purple

**Action buttons:**

**Comment by alinauroz** [comment]
This can be divided into 4 parts

- [ ] Co-creators can open buckets for funding settings and UI changes
- [ ] Co-creators can edit their bucket during funding and UI changes
- [ ] Bucket lifecycle changes
- [ ] Infobars

**Comment by francesca-sp** [comment]
Started testing, and it seems to have been forgotten to update the name of the first status label on the bucket overview page. The status "draft" shows up correctly on bucket pages, but still shows as "unpublished" on the bucket overview page. 

<img width="909" alt="image" src="https://user-images.githubusercontent.com/3641828/215738800-b7260347-2711-4f2e-ba88-3ba15a42115c.png">

**Comment by francesca-sp** [comment]
Secondly, there are errors with the colors of the labels and some text (which are showing up both on the bucket pages, as well as the bucket overview pages): 
* "Idea" is still red, and is supposed to be **yellow** 
* the third status label is supposed to be called "**Funding**", not "funding open", and the color needs to be blue, not red.
* "Funded" is also still red, and is supposed to be **green**
* "Completed" is also still red, and is supposed to be "purple"

**Comment by francesca-sp** [comment]
Thirdly, it seems that 3 different pop-up types/designs are being used for the alerts around the statuses. Could we use one of them for all steps? Especially the ones that show as an undesigned browser pop-up are easy to overlook and also sometimes blocked by add blockers. 

My preference would be this format for the pop-ups:
<img width="544" alt="image" src="https://user-images.githubusercontent.com/3641828/215746324-a47f176d-d9ed-4369-b3aa-fb26a51067f5.png">

**Comment by francesca-sp** [comment]
In the status "completed", the option to cancel the bucket is shown as available, but that option should no longer appear. When you currently try to cancel the bucket once it has been "completed", you get an error message (which makes sense, but hence the option should not even be shown). 

<img width="903" alt="image" src="https://user-images.githubusercontent.com/3641828/215747200-29858816-5b54-4d36-b01c-96788c4e5d12.png">

**Comment by francesca-sp** [comment]
Design element missing: the "Next steps: TIP" piece of this has not been implemented. 

<img width="617" alt="image" src="https://user-images.githubusercontent.com/3641828/215750530-3563a7fa-be80-43b6-a657-24b2c5b2e68f.png">

**Comment by francesca-sp** [comment]
Related issue: https://github.com/cobudget/cobudget/issues/946

**Comment by alinauroz** [comment]
> francesca-sp

should we open a separate issue to fix this?

@aerugo

**Comment by aerugo** [comment]
> should we open a separate issue to fix this?

Yes, that's a good idea @alinauroz

**Comment by aerugo** [comment]
@alinauroz - I have done some testing now, and from what I can see, most issues are resolved. However, the label colors are still not right. See Fran's comment above. Change the label colors as follows:

Draft: Grey
Idea: Yellow
Funding: Blue
Funded: Green
Completed: Purple
